### PR TITLE
pkg/cfgstruct: tie defaults to releases

### PR DESCRIPTION
**Pull request by jtolds**
this change makes it so that by default, the flag defaults are chosen based on whether the build was built as a release build or an ordinary build (be sure to check out scripts/release.sh for how release builds work).

release builds by default get release defaults, whereas ordinary builds by default get dev defaults.

any binary can have its defaults changed by specifying

 --defaults=dev

or

 --defaults=release

Also adds a commit to make release defaults more clear, which extends cfgstruct structs to support either a 'default' tag, or a pair of 'devDefault' and 'releaseDefault' tags, but not both, for added clarity

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Could the PR be broken into smaller PRs?
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?

### CODE REVIEW

File: cmd/identity/batch.go [hunk 1]
```diff
@@ -46,13 +46,13 @@ var (
 		Concurrency   int    `help:"worker concurrency" default:"4"`
 		OutputDir     string `help:"output directory to place keys" default:"."`
 	}
-	isDev bool
+	defaults cfgstruct.BindOpt
```
Comment (mniewrzal) [inline cmd/identity/batch.go]: maybe it can be just local var in `init`?

File: cmd/uplink/cmd/setup.go [hunk 2]
```diff
@@ -31,7 +31,7 @@ var (
 	setupCfg    UplinkFlags
 	confDir     string
 	identityDir string
-	isDev       bool
+	defaults    cfgstruct.BindOpt
```
Comment (mniewrzal) [inline cmd/uplink/cmd/setup.go]: maybe it can be just local var in `init`?

File: cmd/identity/batch.go (hunk 1 above)
Comment (jtolds) [reply cmd/identity/batch.go]: ha, i tried, but this variable is used by multiple init methods in multiple files. i could combine all the init methods if that's preferred

File: cmd/uplink/cmd/setup.go (hunk 2 above)
Comment (jtolds) [reply cmd/uplink/cmd/setup.go]: same here

File: cmd/identity/batch.go (hunk 1 above)
Comment (mniewrzal) [reply cmd/identity/batch.go]: no need, it was just nit

**Pull request merged**